packages/react-native-performance/ios/RNPerformanceManager.m

```diff
@@ -0,0 +1,117 @@
+#import "RNPerformanceManager.h"
+#import <sys/sysctl.h>
+#include <math.h>
+#import <QuartzCore/QuartzCore.h>
+#import <React/RCTRootView.h>
+#import <React/RCTPerformanceLogger.h>
+
+static CFTimeInterval getProcessStartTime() {
+    size_t len = 4;
+    int mib[len];
+    struct kinfo_proc kp;
+
+    sysctlnametomib("kern.proc.pid", mib, &len);
+    mib[3] = getpid();
+    len = sizeof(kp);
+    sysctl(mib, 4, &kp, &len, NULL, 0);
+
+    struct timeval startTime = kp.kp_proc.p_un.__p_starttime;
+    return startTime.tv_sec + startTime.tv_usec / 1e6;
+}
+
+static CFTimeInterval sNativeLaunchStart;
+static CFTimeInterval sNativeLaunchEnd;
+
+@implementation RNPerformanceManager
+{
+    bool hasListeners;
+}
+
+RCT_EXPORT_MODULE();
+
++ (void) initialize
+{
+    [super initialize];
+    CFTimeInterval absoluteTimeToRelativeTime =  CACurrentMediaTime() - [NSDate date].timeIntervalSince1970;
+    sNativeLaunchStart = getProcessStartTime() + absoluteTimeToRelativeTime;
+    sNativeLaunchEnd = CACurrentMediaTime();
+}
+
+- (void)setBridge:(RCTBridge *)bridge
+{
+    [super setBridge:bridge];
+    NSNotificationCenter *notificationCenter = NSNotificationCenter.defaultCenter;
+    [notificationCenter addObserver:self
+                           selector:@selector(contentDidAppear)
+                               name:RCTContentDidAppearNotification
+                             object:nil];
+}
+
+- (void)contentDidAppear
+{
+    CFTimeInterval startTime = CACurrentMediaTime();
```

Comment:
No buffered implementation here?

Reply:
It assumes listener has been set up as a side effect of importing the package. The alternative would be to have a setup function the consumer would call explicitly at some point. While I would agree that's probably technically better, it would be a bit more cumbersome to use the library.